src/api-service/__app__/onefuzzlib/job_templates/templates.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+from typing import List, Optional, Tuple
+
+from onefuzztypes.enums import ErrorCode
+from onefuzztypes.job_templates import JobTemplateConfig
+from onefuzztypes.job_templates import JobTemplateIndex as BASE_INDEX
+from onefuzztypes.job_templates import JobTemplateRequest
+from onefuzztypes.models import Error, Result, UserInfo
+
+from ..jobs import Job
+from ..notifications.main import Notification
+from ..orm import ORMMixin
+from ..tasks.config import TaskConfigError, check_config
+from ..tasks.main import Task
+from .default_templates import TEMPLATES
+from .render import build_input_config, render
+
+
+class JobTemplateIndex(BASE_INDEX, ORMMixin):
+    @classmethod
+    def key_fields(cls) -> Tuple[str, Optional[str]]:
+        return ("name", None)
+
+    @classmethod
+    def get_index(cls) -> List[BASE_INDEX]:
+        entries = [BASE_INDEX(name=x.name, template=x.template) for x in cls.search()]
+
+        # if the local install has replaced the built-in templates, skip over them
+        for name, template in TEMPLATES.items():
+            if any(x.name == name for x in entries):
+                continue
+            entries.append(BASE_INDEX(name=name, template=template))
+
+        return entries
+
+    @classmethod
+    def get_configs(cls) -> List[JobTemplateConfig]:
+        configs = [build_input_config(x.name, x.template) for x in cls.get_index()]
+
+        return configs
+
+    @classmethod
+    def execute(cls, request: JobTemplateRequest, user_info: UserInfo) -> Result[Job]:
+        index = cls.get(request.name)
+        if index is None:
+            if request.name not in TEMPLATES:
+                return Error(
+                    code=ErrorCode.INVALID_REQUEST,
+                    errors=["no such template: %s" % request.name],
+                )
+            template = TEMPLATES[request.name]
+        else:
+            template = index.template
+
+        template = render(request, template)
```

Comment:
a render error here should be included in the return type so we have the opportunity to return it to the caller of `POST job_template`

Reply:
Agreed, see previous comment.